cpp/include/rapidsmpf/communicator/mpi.hpp

```diff
@@ -103,8 +103,9 @@ class MPI final : public Communicator {
      * @brief Construct an MPI communicator.
      *
      * @param comm The MPI communicator to be used for communication.
+     * @param options Configuration options.
      */
-    MPI(MPI_Comm comm);
+    MPI(MPI_Comm comm, config::Options options);
```

Comment:
@madsbk 
If I'm not mistaken, `Options` is more or less a singleton, and not expected to change during the execution. So, can we use a `shared_ptr` rather than pass by value?

Reply:
Actually, `config::Options` is a shared pointer to `config::OptionsImpl` in order to avoid having to use `shared_ptr` everywhere :)